### DISCUSSION TITLE
Sonoff ZBDongle-E 6.10.3 and Z2M.

### DISCUSSION THREAD

**Discussion by JimmyJimbolo** [Koenkk/zigbee2mqtt#23765] [Devices]
Hello.

I have a Sonoff ZBDongle-E (V2 model, EFR32MG21) with firmware version 6.10.3.0 build 297. It works fine on ZHA, but since I've been reading about how good Z2M is for a long time, I wanted to try it out.

My first tests have been failure after failure. I can't find clear documentation specifying what configuration I have to put in the Zigbee2Mqtt addon and what the zigbee2mqtt/configuration.yaml file has to have. I have to re-read the official Z2M page again.

The addon log returns various errors:

[2024-08-28 18:44:09] info: zh:zstack:znp: Skip bootloader for CC2652/CC1352 [2024-08-28 18:44:27] error: z2m: Error while starting zigbee-herdsman
[2024-08-28 18:44:27] error: z2m: Failed to start zigbee
[2024-08-28 18:44:27] error: z2m: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
[2024-08-28 18:44:27] error: z2m: Exiting...
[2024-08-28 18:44:27] error: z2m: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)

My first question is, do I have to update the dongle firmware to version 7.4.x or 8.0.x?

**Comment by chris-1243** [comment]
First and if testing on the same computer, you must completely stop ZHA. I would recommand a full backup as well.

Then, I would flash your dongle-E to 7.4.3 (8.x.x are still in dev phase).

> [2024-08-28 18:44:09] info: zh:zstack:znp: Skip bootloader for CC2652/CC1352

As soon as those steps are done and in `configuration.yaml`

```
serial: 
  adapter: ember
```

https://www.zigbee2mqtt.io/guide/adapters/emberznet.html

**Reply by JimmyJimbolo** [reply]
Hi. Thanks for answering.

After calming down a bit and finding a video where Zigbee2MQTT was up and running in 5 minutes, I decided to start from the beginning and uninstall Z2M and follow the steps in the video. I finally found a video that says something can be done in 5 minutes, AND THEN IT'S TRUE!! Now Z2M works for me.

I think the error was in the Z2M mqtt configuration. It's possible that I didn't have the server data set correctly.

I will follow your advice to update the Sonoff driver to version 7.4.3 mainly to switch to ember. I have found several ways to do it, via WEB, vis soft... any advice on how to do it better?

Thanks again.

**Reply by chris-1243** [reply]
Have a look there

https://github.com/darkxst/silabs-firmware-builder

I used this repo for mine and it is working flawlessly. (If you have Homeassistant, there is normally an addon of this tool...)

You may also try this tool to flash your Dongle-E

https://github.com/Nerivec/ember-zli

**Reply by JimmyJimbolo** [reply]
Hi.

It's already updated super easy and using ember. I don't know what benefits I'll get but I'll find out.

Thank you very much chris-1234
